### ISSUE TITLE
ignores externals

### ISSUE THREAD

**Issue by gaje** [DanielSchaffer/webpack-babel-multi-target-plugin#3]
If you have any modules defined in webpack.config as `externals`, for some reason the inclusion of this plugin is causing webpack to try bundle them. I tried adding the same references in my `externals` to both the `exclude` array and also the `doNotTarget` array to see if they would be ignored, but webpack is dying on trying to import these even though they are defined as externals. 

If i remove this plugin, webpack behave as expected so i feel the issue exists internal here.

**Comment by DanielSchaffer** [comment]
I was able to reproduce the issue with externals - the reason is that in order to do what it does, the plugin actually changes the full path of the request of each module - so, for example, an `import * as $ from 'jquery'` effectively becomes `import * as $ from 'jquery?babel-target=legacy'` and  `import * as $ from 'jquery?babel-target=modern'`. Once that happens, the key you have in your externals configuration no longer matches, assuming you've got something like this:

```
 externals: {
        'jquery': {
            amd: 'jQuery',
            commonjs: 'jQuery',
            root: '$',
        },
}
```

For example, what _does_ work, for the purposes of demonstration, is if you change it to use the new targeted requests generated by the plugins:

```
        'jquery?babel-target=legacy': {
            amd: 'jQuery',
            commonjs: 'jQuery',
            root: '$',
        },
        'jquery?babel-target=modern': {
            amd: 'jQuery',
            commonjs: 'jQuery',
            root: '$',
        }
```

I was not able to reproduce the problem of adding configuration `doNotTarget` - once I added `jquery` to the `doNotTarget` option, it worked correctly:

```
    ....
    externals: {
        'jquery': {
            amd: 'jQuery',
            commonjs: 'jQuery',
            root: '$',
        },
    },
    ...
   plugins: [
       new BabelMultiTargetPlugin({
            doNotTarget: [
                /jquery/,
            ],
        }),
   ],
```

That said, you obviously shouldn't have to do _any_ of that, so I'll add a patch to check externals so you'll get the expected behavior.

**Comment by DanielSchaffer** [comment]
This should be fixed in [v1.0.0-alpha.6](https://github.com/DanielSchaffer/webpack-babel-multi-target-plugin/releases/tag/v1.0.0-alpha.6) with the caveat that it can't currently support [externals defined as a function](https://webpack.js.org/configuration/externals/#function) due to the way the webpack hooks are set up. The part that does the check currently runs as part of a synchronous hook, and the functions themselves are designed to be asynchronous, so I'd need to either figure out if there's somewhere the external state has already been evaluated by webpack, or find an asynchronous hook to use with the right timing to evaluate the externals.
